examples/tv-casting-app/darwin/TvCasting/MatterBridge/DiscoveredNodeData.mm

```diff
@@ -0,0 +1,76 @@
+/**
+ *
+ *    Copyright (c) 2020-2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "DiscoveredNodeData.h"
+#include <lib/dnssd/Resolver.h>
+
+@implementation DiscoveredNodeData
+
+- (DiscoveredNodeData *)initWithChipDiscoveredNodeData:(void *)chipDiscoveredNodedata
+{
+    self = [super init];
+    if (self) {
+        chip::Dnssd::DiscoveredNodeData * data = (chip::Dnssd::DiscoveredNodeData *) chipDiscoveredNodedata;
```

Comment:
We should just have the right type coming in, if possible... This cast is extremely dangerous.

Reply:
Done. Avoided the cast by doing the conversion in a separate C++ class: DiscoveredNodeDataConverter